File: 이건희/level1/성격 유형 검사하기.py
def solution(survey, choices):
    result = ""
    category = {"R" : 0, "T" : 0, "C" : 0, "F" : 0, "J" : 0, "M" : 0, "A" : 0, "N" : 0}
    
    for i in range(len(survey)):
        if choices[i] == 4:
            continue
        elif choices[i] >= 5:
            score = choices[i] - 4
            category[survey[i][1]] += score
        else:
            score = abs(choices[i] - 4)
            category[survey[i][0]] += score
    
    result += 'R' if category['R'] >= category['T'] else 'T'
    result += 'C' if category['C'] >= category['F'] else 'F'
    result += 'J' if category['J'] >= category['M'] else 'M'
    result += 'A' if category['A'] >= category['N'] else 'N'
    
    return result

File: 이훈일/Programmers/Level1/성격 유형 검사하기.java
class Solution {
    public String solution(String[] survey, int[] choices) {
        String[] pesonality = new String[]{"RT","CF","JM","AN"};
        int[] score = new int[4];

        for(int i = 0; i < survey.length; i++) {
            String[] surveyArr = survey[i].split("");
            String first = surveyArr[0];
            if(first.equals("A") || first.equals("N")){
                if(first.equals("A")) {
                    score[3] += (choices[i]-4) * -1;
                } else{
                    score[3] += choices[i]-4;
                }
            } else if(first.equals("C") || first.equals("F")){
                if(first.equals("C")) {
                    score[1] += (choices[i]-4)* -1;
                } else{
                    score[1] += (choices[i]-4);
                }
            } else if(first.equals("J") || first.equals("M")){
                if(first.equals("J")) {
                    score[2] += (choices[i]-4)*-1;
                } else{
                    score[2] += choices[i] -4;
                }
            } else{
                if(first.equals("R")) {
                    score[0] += (choices[i] - 4) * -1;
                } else{
                    score[0] += choices[i]-4;
                }
            }
        }
        String answer = "";

        for(int i = 0; i < score.length; i++){
            String[] perArr = pesonality[i].split("");
            if(score[i] >= 0){
                answer += perArr[0];
            } else{
                answer += perArr[1];
            }
        }

        return answer;
    }
}
